source/wifi/wifi_hal.c

```diff
@@ -8425,6 +8445,27 @@ int main(int argc,char **argv)
         mac_addr_ntoa(mac_addr,output_struct.wifi_devMacAddress);
         printf("wifi_devMacAddress=%s \t wifi_devAssociatedDeviceAuthentiationState=%d \t, wifi_devSignalStrength=%d \t,wifi_devTxRate=%d \t, wifi_devRxRate =%d \t\n ", mac_addr,output_struct.wifi_devAssociatedDeviceAuthentiationState,output_struct.wifi_devSignalStrength,output_struct.wifi_devTxRate,output_struct.wifi_devRxRate);
     }
+	
+   if(strstr(argv[1],"wifi_startNeighborScan")!=NULL)
+    {
+	int args_index = 0;
+        if(argc <= 5 )
+        {
+    	    printf("Insufficient arguments , usage :: \n");  
+            printf(" wifihal wifi_startNeighborScan [apindex (0-15)] [scanmode (0 to 4)] [dwelltime] [channel_listcount] [ channel-list]... \n");    
+	    exit(-1);
+        }
+        INT apindex = atoi(argv[2]);
+        wifi_neighborScanMode_t scan_mode = atoi(argv[3]);
+        INT dwell_time = atoi(argv[4]);
+        UINT chan_num = atoi(argv[5]);
+	UINT *chan_list = (UINT *) malloc((argc-5)*sizeof(UINT));
+	for( args_index=0 ; args_index < argc-6 ; args_index++ ){
+      	    chan_list[args_index] = (UINT)atoi(argv[args_index+6]);
+        }
+        wifi_startNeighborScan(apindex, scan_mode, dwell_time, chan_num, chan_list);
```

Comment:
please print return code from this wifi hal function

Reply:
done